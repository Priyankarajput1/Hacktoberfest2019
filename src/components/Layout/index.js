import React from 'react';
import { injectGlobal } from 'emotion';
import PropTypes from 'prop-types';
import { pathOr } from 'ramda';
import { useStaticQuery, graphql } from 'gatsby';

import Header from '../Header';

injectGlobal`
  * {
    box-sizing: border-box;
  }

  html, body {
    padding: 0;
    margin: 0;
    font-family: 'Quicksand', sans-serif;
  }
`;

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <>
      <Header
        siteTitle={pathOr('No title', ['site', 'siteMetadata', 'title'], data)}
      />
      <div
        style={{
          margin: '0 auto',
          maxWidth: 960,
          padding: '0px 1.0875rem 1.45rem',
          paddingTop: 0,
        }}
      >
        <main>{children}</main>
        <footer>
          <hr />
          <h3>This is footer</h3>
          <a href="https://www.greenberry.nl">Greenberry</a>
        </footer>
      </div>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
