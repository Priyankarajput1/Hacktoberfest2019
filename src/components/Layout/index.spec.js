import React from 'react';
import renderer from 'react-test-renderer';

import Layout from './index';

describe('components/Layout', () => {
  it('should render', () => {
    const component = renderer.create(
      <Layout>
        <h1>Hello layout</h1>
      </Layout>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
