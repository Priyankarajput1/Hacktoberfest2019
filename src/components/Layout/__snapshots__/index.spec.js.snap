// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout should render 1`] = `
Array [
  <header
    style={
      Object {
        "background": "#999",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          No title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <main>
      <h1>
        Hello layout
      </h1>
    </main>
    <footer>
      <hr />
      <h3>
        This is footer
      </h3>
      <a
        href="https://www.greenberry.nl"
      >
        Greenberry
      </a>
    </footer>
  </div>,
]
`;
